src/base/bittorrent/bencoderesumedatastorage.cpp

```diff
@@ -201,11 +206,15 @@ void BitTorrent::BencodeResumeDataStorage::loadQueue(const Path &queueFilename)
         if (rxMatch.hasMatch())
         {
             const auto torrentID = BitTorrent::TorrentID::fromString(rxMatch.captured(1));
-            const int pos = m_registeredTorrents.indexOf(torrentID, start);
+            const qsizetype pos = registeredTorrentsIndexes.value(torrentID, -1);
             if (pos != -1)
             {
-                std::swap(m_registeredTorrents[start], m_registeredTorrents[pos]);
-                ++start;
+                if (pos != queuePos)
+                {
+                    m_registeredTorrents.swapItemsAt(pos, queuePos);
+                    registeredTorrentsIndexes.insert(torrentID, queuePos);
```

Comment:
Well, we really don't need to update the current torrent's position in the index, because we won't need it anymore. But we need to update the position of the torrent that the current one was swapped with.
```suggestion
                    registeredTorrentsIndexes.insert(m_registeredTorrents.at(pos), pos);
```